### ISSUE TITLE
Report Concatenation fails with sparse data

### ISSUE THREAD

**Issue by scottiegarcia** [capitalone/DataProfiler#146]
**General Information:**
 - OS: MacOS Catalina 10.15.7
 - Python version: 3.7.9
 - Library version: 0.4.2

**Describe the bug:**
Report Concatenation fails when missing datetime values in data.

**To Reproduce:**
```
import pandas as pd
from dataprofiler import Profiler
from datetime import datetime

dates = [None] * 20
dates[15] = datetime.strptime("2014-12-18", "%Y-%M-%d").date()
dates[16] = datetime.strptime("2015-07-21", "%Y-%M-%d").date()
dates[19] = datetime.strptime("2018-09-01", "%Y-%M-%d").date()

df = pd.DataFrame({"date": dates})

df_1 = df[:10]
df_2 = df[10:]

profiles_1 = Profiler(data=df_1)

profiles_2 = Profiler(data=df_2)

profiles = profiles_1 + profiles_2 
```

**Expected behavior:**
To be able to handle `None` values in statistics when concatenating reports.

**Additional context:**
This only seems to occur with `datetime.date` objects where the null values in the column remain `None` instead of the numpy `NaT` value. Might be reasonable to expect user to prep data better beforehand, but better handling of None values is probably in order.

Stack trace below
```
Traceback (most recent call last):
  File "bug2.py", line 19, in <module>
    profiles = profiles_1 + profiles_2
  File "/Users/AXY161/Projects/dataprofiler-oss/dataprofiler/profilers/profile_builder.py", line 454, in __add__
    self._profile[profile_name] + other._profile[profile_name]
  File "/Users/AXY161/Projects/dataprofiler-oss/dataprofiler/profilers/profile_builder.py", line 147, in __add__
    self.profiles[profile_name] + other.profiles[profile_name]
  File "/Users/AXY161/Projects/dataprofiler-oss/dataprofiler/profilers/column_profile_compilers.py", line 99, in __add__
    self._profiles[profile_name] + other._profiles[profile_name]
  File "/Users/AXY161/Projects/dataprofiler-oss/dataprofiler/profilers/datetime_column_profile.py", line 79, in __add__
    if other._dt_obj_min is None or self._dt_obj_min < other._dt_obj_min:
TypeError: '<' not supported between instances of 'NoneType' and 'Timestamp'  
```

**Comment by JGSweets** [comment]
I will be fixing shortly, thanks for the bug find!

**Comment by JGSweets** [comment]
https://github.com/capitalone/DataProfiler/pull/149

**Comment by lettergram** [comment]
Will be released v0.4.3
